### PR TITLE
Subpixel

### PR DESCRIPTION
**Pull request by EderSantana**
let me run tests on travis. they behave differently when I run them on my computer.

This PR is the backend part for Subpixel CNNs in Keras. Regular Convs + Lambda and this backend can already get us there. But I'm thinking about building a proper layer to handle shape inference etc.

### CODE REVIEW

File: examples/variational_autoencoder_deconv.py [hunk 1]
```diff
@@ -13,6 +13,8 @@
 from keras import objectives
 from keras.datasets import mnist
 
+K.set_image_dim_ordering('th') # this is a Theano oriented example
```
Comment (fchollet) [inline examples/variational_autoencoder_deconv.py]: It seems you have included with this PR code that shouldn't be there (or may it's just this one line)? I made this example compatible with both dim orderings a while ago.

**Comment by EderSantana** [comment]
yeah, I will rebase my PR. but the problem I was having was that the TH code I found online is not quite similar to TF depth_to_space. I gotta look at that carefully
